src/core/symbols/line_symbol.cpp

```diff
@@ -1127,6 +1131,31 @@ SplitPathCoord LineSymbol::createDashGroups(
 			bool is_first_dashgroup = dashgroup == 1;
 			bool is_last_dashgroup  = dashgroup == num_dashgroups;
 			
+			if (mid_symbols == CenterOfDashGroup)
+			{
+				auto position = cur_length;
+				if (is_last_dashgroup && half_last_group)
+				{
+					position = end.clen;
+				}
+				else if (!(is_first_dashgroup && half_first_group))
+				{
+					position += (adjusted_dash_length * dashes_in_group
+					             + in_group_break_length_f * (dashes_in_group - 1)) / 2;
+				}
+				position -= (mid_symbol_distance_f * (mid_symbols_per_spot - 1)) / 2;
+				auto split = dash_start;
+				for (int i = 0; i < mid_symbols_per_spot && position <= end.clen; ++i, position += mid_symbol_distance_f)
+				{
+					if (position <= start.clen)
+						continue;
+					split = SplitPathCoord::at(position, split);
+					if (mid_symbol_rotatable)
+						orientation = split.tangentVector().angle();
+					mid_symbol->createRenderablesScaled(split.pos, orientation, output);
+				}
+			}
+			
```

Comment:
This line, and a few others, introduces whitespace error. While neither the coding style nor GitHub tools mandate trailing-whitespace free code, `git show --check origin/linesymbol-extended` shows them prominently.

Reply:
Apart from how it is judged by `git show --check`, I fail to see this as an error, or even as *trailing* whitespace. Actually it is *leading* whitespace, consistent with the lines before and after. Git can match this only with blank lines having the same amount of whitespace, not with any blank line. I feel like that is what I want.

But I do not have a strong opinion about this. Qt seems to use strict blank lines, but I don't find an explicit rule about this (unless you interprete "blank line" in a very strict sense).

Reply:
Let's quit the holy war before it breaks out. I've instructed Git not to highlight trailing white space. And a blank line on my side of the wire matches regexp `^$`. :)